### PR TITLE
feat(TOTP): use feature toggle via authserver.conf

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
## Changes Proposed:
-  Restore TOTP from @Winfidonarleyan previously reverted because it had a bug ( original implementation: https://github.com/azerothcore/azerothcore-wotlk/pull/5640 )
- NOTE: I did **NOT** fix that bug
-  Made the system configurable and disabled by default, so one can just feature-toggle it
- Basically my only addition to the original @Winfidonarleyan work is: https://github.com/azerothcore/azerothcore-wotlk/commit/a2f88ab9773294a86bb0c692401a808c1250c37e

## Issues Addressed:

- Related to #5977 #5970 #5640 #5971

## Tests Performed:

- Logged in-game successfully

## How to Test the Changes:

1. Try to access the game

## Known Issues and TODO List:

- The TOTP bug should be fixed in another PR, for now it's enough to keep it disabled

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
Guess you can close https://github.com/azerothcore/azerothcore-wotlk/pull/5972

**Comment by Kitzunu** [comment]
And the bug is already fixed with robotos commit

**Comment by FrancescoBorzi** [comment]
@Kitzunu unfortunately I wasn't able to test the TOTP itself so I can't tell you that myself

File: src/server/authserver/authserver.conf.dist [hunk 1]
```diff
@@ -185,7 +185,15 @@ LoginDatabase.SynchThreads = 1
 
 ###################################################################################################
 # CRYPTOGRAPHY
+
+#    EnableTOTP
```
Comment (Kitzunu) [inline src/server/authserver/authserver.conf.dist:189]:
```suggestion
#
#    EnableTOTP
```

To follow standard

**Comment by locus313** [comment]
Thanks, @FrancescoBorzi. I will test this in the next couple of hours with accounts that have TOTP enabled and disabled.

**Comment by locus313** [comment]
Just finished testing the following:
Account test1 with totp enabled 
Account test2 with totp disabled

- EnableTOTP = 0
   - test1
      - no prompt
   - test2 
      - no prompt 
      
- EnableTOTP = 1
   - test1
      - prompted for code
      - .acccount set 2fa test1 off
      - no prompt
   - test2 
      - no prompt 
      - .acccount set 2fa test1 <totp_secret>
      - prompted for code


Think we are good to merge this now.

**Comment by FrancescoBorzi** [comment]
@locus313 thank you very much for testing this!

Need 1 approval :)

**Review by locus313 — APPROVED** [review]

**Review by r-o-b-o-t-o — APPROVED** [review]

**Comment by Kitzunu** [comment]
Thanks

**Comment by Winfidonarleyan** [comment]
Very goos, thanks for pr

**Pull request merged**